app/Main.hs

```diff
@@ -44,7 +44,11 @@ readJSON = fmap eitherDecode . BS.readFile
 
 
 readYAML :: FilePath -> IO (Either String Value)
-readYAML = fmap decodeEither . B.readFile
+readYAML filepath = do
+  result <- fmap decodeEither' $ B.readFile filepath
+  case result of
+    Left err  -> pure $ Left (show err)
+    Right val -> pure $ Right val
```

Comment:
This whole section could be simplified to `fmap (first show . decodeEither') . B.readFile` where `first` is imported from `Data.Bifunctor`

If you could just make that small adjustment, then I'll be very happy to merge this PR. 
Thank you for your contribution 🙂

Reply:
I know this could have been simplified but i didn't know how. I actually searched for it on stackoverflow and found the solution with befunctor. But instead i thought a bit longer line without extra import was maybe more readable code. Anyway i can make this adjustment np

Reply:
Bifunctor is still pretty new, so that doesn't surprise me :wink:. As for the extra import, since this is coming from `base` anyway I think it's fine.